reference/reference.py

```diff
@@ -335,34 +313,34 @@ def sign(secnonce: bytearray, secshare: bytes, my_id: int, session_ctx: SessionC
     if not 0 <= my_id < 2**32:
         raise ValueError('The signer\'s participant identifier is out of range')
     (Q, gacc, _, b, R, e) = get_session_values(session_ctx)
-    k_1_ = int_from_bytes(secnonce[0:32])
-    k_2_ = int_from_bytes(secnonce[32:64])
+    k_1_ = Scalar.from_bytes_checked(secnonce[0:32])
```

Comment:
We might want to wrap this in a try-except block. Secp256k1lab internaly raises a plain `ValueError`, but in our case, we’d prefer to raise `ValueError("first secnonce value is out of range")` instead. Something like this would be nice:
```
try:
    k1 = Scalar.from_bytes_checked(x)
    if k1 == 0:           # treat zero exactly like any other bad input
        raise ValueError
except ValueError:
    raise ValueError("first secnonce value is out of range.")
```

Reply:
Took your suggestion and added the use of `Scalar.from_bytes_nonzero_checked` as a future TODO, since this constructor is not available in secp256k1lab yet.